pkg/applier/applier.go

```diff
@@ -42,19 +54,32 @@ func NewApplier(
 	owner metav1.Object,
 	scheme *runtime.Scheme,
 	merger Merger,
+	applierOptions *ApplierOptions,
 ) (*Applier, error) {
 	if templateProcessor == nil {
 		return nil, goerr.New("applier is nil")
 	}
 	if client == nil {
 		return nil, goerr.New("client is nil")
 	}
+	if applierOptions == nil {
+		applierOptions = &ApplierOptions{}
+	}
+	if applierOptions.Backoff == nil {
+		applierOptions.Backoff = &wait.Backoff{
```

Comment:
not sure, so what's the worse case when fail to apply, 100,500,2500,12500,72500 ms? shall we add a cap here? and not sure will it block reconcile or it only used in executable? feel in reconcile we want method not blocking but reply at once

Reply:
Replaced by retry.defaultBackoff and add a custom one in the cli.